src/components/Applications.jsx

```diff
@@ -36,10 +36,19 @@ const LoadingAppTiles = memo(({ num }) => {
 })
 LoadingAppTiles.displayName = LoadingAppTiles
 
-export const Applications = () => {
+export const Applications = ({ onAppsFetched }) => {
   const showLogout = !!flag('home.mainlist.show-logout')
   const shortcuts = useHomeShortcuts()
-  const { data, fetchStatus } = useQuery(appsConn.query, appsConn)
+  const { data, fetchStatus, lastFetch } = useQuery(appsConn.query, appsConn)
+  const hasFetched = useRef(false)
+
+  useEffect(() => {
+    !hasFetched.current &&
+      fetchStatus === 'loaded' &&
+      lastFetch &&
+      onAppsFetched() &&
+      (hasFetched.current = true)
```

Comment:
nit: I very very dislike this kind of writing. I find that: 

```
if(!hasFetched.current && fetchStatus === 'loaded' && lastFetch){
openAppsFetched() 
hasFetched.current = true
}
```

so much more readable!